system/jlib/jflz.cpp

```diff
@@ -723,16 +728,30 @@ class jlib_decl CFastLZCompressor : public CInterface, public ICompressor
 
     size32_t write(const void *buf,size32_t len)
     {
-        if (len+inlen>inmax) {
-            if (trailing)
-                return 0;
-            flushcommitted();
-            if (len+inlen>inmax) 
-                len = inmax-inlen;
+        // no more than wrmax per write
+        size32_t lenb = wrmax;
+        byte *b = (byte *)buf;
+        size32_t written = 0;
+        while (len)
+        {
+            if (len < lenb)
+                lenb = len;
+            if (lenb+inlen>inmax) {
+                if (trailing)
+                    return written;
+                flushcommitted();
```

Comment:
@mckellyln Wouldn't it be worth trying to fill inbuff up to the limit of inmax before compressing and flushing?